tests/integration/test_charm.py

```diff
@@ -26,17 +28,23 @@
     reason="skipping deploy, model expected to be provided.",
 )
 @pytest.mark.abort_on_fail
-async def test_build_and_deploy(ops_test: OpsTest):
+async def test_deploy(ops_test: OpsTest, postgresql_charm, postgresql_tester_charm):
     """Build the charm-under-test and deploy it.
 
     Assert on the unit status before any relations/configurations take place.
     """
-    # Build and deploy charm from local source folder.
-    charm = await ops_test.build_charm(".")
+    # Deploy both PostgreSQL and PostgreSQL test charms.
     resources = {"postgresql-image": METADATA["resources"]["postgresql-image"]["upstream-source"]}
     await ops_test.model.deploy(
-        charm, resources=resources, application_name=APP_NAME, num_units=len(UNIT_IDS), trust=True
+        postgresql_charm,
+        resources=resources,
+        application_name=APP_NAME,
+        num_units=len(UNIT_IDS),
+        trust=True,
     )
+
+    await ops_test.model.deploy(postgresql_tester_charm, application_name=TESTER_APP_NAME)
```

Comment:
To make this part run a little faster, you could use [`asyncio.gather()`](https://docs.python.org/3/library/asyncio-task.html#running-tasks-concurrently) to gather these functions:

```python
await asyncio.gather(
    ops_test.model.deploy(
        charm, resources=resources, application_name=APP_NAME, num_units=len(UNIT_IDS), trust=True
        postgresql_charm,
        resources=resources,
        application_name=APP_NAME,
        num_units=len(UNIT_IDS),
        trust=True,
    ),
    ops_test.model.deploy(postgresql_tester_charm, application_name=TESTER_APP_NAME)
)
```